exporters/tests/test_failure_exporter_pager_duty.py

```diff
@@ -0,0 +1,201 @@
+#!/usr/bin/env python3
+#
+# Copyright Red Hat
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+#
+
+import os
+from contextlib import nullcontext
+from typing import Optional
+
+import pytest
+from prometheus_client.core import REGISTRY
+
+from failure.collector_pagerduty import PagerdutyFailureCollector
+from pelorus.errors import FailureProviderAuthenticationError
+
+PAGER_DUTY_TOKEN = os.environ.get("PAGER_DUTY_TOKEN")
+
+
+def setup_pager_duty_collector(
+    token: str = "fake_token",
+    incident_urgency: Optional[str] = None,
+    incident_priority: Optional[str] = None,
+) -> PagerdutyFailureCollector:
+    return PagerdutyFailureCollector(
+        token=token,
+        incident_urgency=incident_urgency,
+        incident_priority=incident_priority,
+    )
+
+
+@pytest.mark.integration
+def test_pager_duty_connection():
+    collector = setup_pager_duty_collector()
+    with pytest.raises(FailureProviderAuthenticationError) as auth_error:
+        collector.search_issues()
+    assert "Check the TOKEN: not authorized, invalid credentials" in str(
+        auth_error.value
+    )
+
+
+@pytest.mark.integration
+@pytest.mark.skipif(
+    not PAGER_DUTY_TOKEN,
+    reason="No PagerDuty token set, run export PAGER_DUTY_TOKEN=token",
+)
+def test_pager_duty_search():
+    collector = setup_pager_duty_collector(PAGER_DUTY_TOKEN)
+
+    with nullcontext() as context:
+        issues = collector.search_issues()
+
+    assert context is None
+    assert len(issues) == 7
```

Comment:
I did the most easy way https://developer.pagerduty.com/docs/ZG9jOjExMDI5NTU4-pagination
but if user has more than the limit 10000, it will fail. Should we handle this case or its hard that user will have so much incidents?

Reply:
Is this per API or per service?

Reply:
Please create an issue so we don't forget this limit and resolve this conversation after with pointer to the issue.

Reply:
Created issue https://github.com/konveyor/pelorus/issues/853, closing the conversation as we agreed on resolving the problem in other PR